src/server/scripts/Outland/CoilfangReservoir/TheUnderbog/boss_hungarfen.cpp

```diff
@@ -159,8 +157,36 @@ struct npc_underbog_mushroom : public ScriptedAI
     uint32 _counter;
 };
 
+// 31693 - Putrid Mushroom Primer
+class spell_hungarfen_putrid_mushroom_primer : public SpellScript
+{
+    PrepareSpellScript(spell_hungarfen_putrid_mushroom_primer);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SUMMON_UNDERBOG_MUSHROOM });
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        Creature* caster = GetCaster()->ToCreature();
+        if (!caster || !caster->IsAIEnabled())
```

Comment:
as far as I'm concern, there's no need to check for IsAIEnabled() to access AI class ever since Jackpot introduced AI changing status where AI is never null.

Reply:
ok, thanks